apps/studio/components/interfaces/Settings/Vault/VaultToggle.tsx

```diff
@@ -1,27 +1,50 @@
 import * as Tooltip from '@radix-ui/react-tooltip'
 import { PermissionAction } from '@supabase/shared-types/out/constants'
 import { useParams } from 'common'
-import { useTheme } from 'next-themes'
 import { observer } from 'mobx-react-lite'
+import { useTheme } from 'next-themes'
 import Link from 'next/link'
 import { useState } from 'react'
+import toast from 'react-hot-toast'
 import { Button, IconExternalLink } from 'ui'
 
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import { useDatabaseExtensionEnableMutation } from 'data/database-extensions/database-extension-enable-mutation'
+import { useDatabaseExtensionsQuery } from 'data/database-extensions/database-extensions-query'
 import { useCheckPermissions, useStore } from 'hooks'
 import { BASE_PATH } from 'lib/constants'
 
 const VaultToggle = () => {
-  const { meta, ui } = useStore()
   const { ref } = useParams()
+  const { meta, ui } = useStore()
   const { resolvedTheme } = useTheme()
+  const { project } = useProjectContext()
   const [isEnabling, setIsEnabling] = useState(false)
   const canToggleVault = useCheckPermissions(PermissionAction.TENANT_SQL_ADMIN_WRITE, 'extensions')
 
-  const vaultExtension = meta.extensions.byId('supabase_vault')
+  const { data } = useDatabaseExtensionsQuery({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+  })
+  const vaultExtension = (data ?? []).find((ext) => ext.name === 'supabase_vault')
   const isNotAvailable = vaultExtension === undefined
 
+  const { mutate: enableExtension } = useDatabaseExtensionEnableMutation({
+    onSuccess: () => {
+      toast.success(`Vault is now enabled for your project!`)
+    },
+    onError: (error) => {
+      toast.error(`Failed to enable Vault for your project: ${error.message}`)
+      setIsEnabling(false)
+    },
+  })
+
   const onEnableVault = async () => {
     if (vaultExtension === undefined) return
+    if (project === undefined) return console.error('Project is required')
+    if (project.connectionString === undefined)
+      return console.error('Connection string is required')
```

Comment:
Vault is only supported on platform, so this is fine, right?

Also, this is entirely personal preference (so feel free to ignore) but if an if statement is on multiple lines it freaks me out if it's missing `{}` 😅

Reply:
i think vault should also be on self hosted actually as vault is just a database extension - yeah we dont gate Vault in the project settings nav too